### DISCUSSION TITLE
photos are not displayed in the correct size ratio

### DISCUSSION THREAD

**Discussion by FA1980** [photoprism/photoprism#4444] [Q&A]
In my case, the photos are not displayed in the correct size ratio in the photo view, they are always compressed or stretched. In the thumbnail folder, however, the photos are in the correct ratio. I am grateful for solutions.

Kind regards
Falk

**Comment by lastzero** [comment]
In order for a community member to be able to help you, it would be very helpful to know the version you are using, the file types affected and the runtime environment (Linux with Docker, Windows,...).

**Reply by ad08fee3** [reply]
Hi @lastzero,
Fellow programmer here. I'm also seeing this issue. 

To be more clear about what I am seeing: When clicking on a photo to show it in the photo view, the image appears to be stretched horizontally. 

On my desktop PC (Windows) I have noticed that changing the size of the browser window can "fix" the issue. When I initially click an image and open the photo viewer, it appears stretched. However when I shrink the browser window far enough, the image "pops" and is no longer stretched. It returns back to its normal ratio once the browser window is small enough. Oddly enough, this is not affected by page zoom level at all. It will only ever "pop" back to the normal ratio if the browser window is physically adjusted to a small enough size that the photo must scale down.

Experimenting with this lead to an interesting observation: The height and width of the image is the correct size and ratio, but it appears that both the right and left edges of the photo are cropped off and then the image is stretched horizontally so that the cropped image is the expected ratio again.

For a physical analogy, imagine you have two copies of a photo printed out on stretchy elastic paper. You cut off about 10% of the right and left edges of one of the photos, and then stretch the photo so the cropped photo is the same size as the uncropped photo.

It looks like there is a funky bug in the image scaling logic.

Please take a look at the included example video:

https://github.com/user-attachments/assets/ad14d552-e3b5-4019-b11c-b0f3fc75aab6

Pay special attention to the edge of the building on the left and the number of windows on the building on the right. You can see it gets cropped, but the outline/border/size of the image remains the same. I also demo how page zoom does not affect it.

---

According to my settings page, I see:

>**PhotoPrism® CE** 
>Build 240711-2197af848

I'm currently running PhotoPrism in a Docker container on a desktop PC/server running Unraid OS.

I see the issue where images are stretched on both my Windows computer (in both Firefox and Chrome) along with accessing the site on my mobile phone (Firefox on Android).

On Firefox on Android I can pinch to zoom and move the photo around, but I cannot make the image pop back to normal since I cannot adjust the window size on mobile.

Hopefully this helps! I'd be happy to get you log dumps or anything, if you'd let me know where to grab those.

**Reply by lastzero** [reply]
I don't see this happening with Chrome on Linux. Could you test whether the behavior is specific to a particular browser type, version, or operating system? Does reloading the tab after resizing the browser window fix the issue?

**Reply by ad08fee3** [reply]
I just tested this on my Linux laptop in Firefox and confirmed it was happening there as well.

To see if reloading would fix it, I shrank the window down to a small size so the photo "popped" back to its proper ratio. I reloaded the tab, and it put me back at the album view. I clicked the photo again and it was rendered properly, as expected. But when I expanded the window again it "popped" back out to the skewed size once the window was large enough.

It looks like it doesn't matter which OS or browser I use, unfortunately.
I haven't experimented with changing the version of Photoprism yet.

**Reply by graciousgrey** [reply]
Thanks for testing! Does this happen to all photos or only to some special size ratios? Can you reproduce this with one of the photos on our public demo:

https://demo.photoprism.app/library/browse

**Reply by lastzero** [reply]
@mozilla2012 If reloading the page fixes the problem, it could be that the window size that the browser reports to the application does not match the actual window size:

- Which browsers and operating systems have you tried?
- Do you have any browser plugins installed, for example https://jshelter.org/?

**Reply by ad08fee3** [reply]
> Does this happen to all photos or only to some special size ratios? 

It's happened with 100% of the photos that I have tested. I'm not about to test all 33,000 photos on my server though 🙂

> Can you reproduce this with one of the photos on our public demo

Oh good idea! I can confirm that this is _not_ happening on the public demo! I have tried in both Firefox and Chrome on Windows and am unable to reproduce it on the public demo.

> If reloading the page fixes the problem

To be clear, reloading the page did _not_ fix the problem.

> Which browsers and operating systems have you tried?

Firefox on Linux (Gallium OS, aka Ubuntu), and Firefox/Chrome/Edge on Windows 10.

I do have lots of browser plugins installed, but this issue manifests itself in both Incognito mode and regular browsers. In Edge I have zero plugins/extensions installed at all and it still occurred.

---

I do have an update though! I found something intriguing.
When looking at the Demo server, I noticed the image was "reloading" as I resized the browser window.
If I right-clicked --> "Open image in a new tab" I would get the following URLs and image sizes. 
As I shrank the window, the image resolution would change as well:
1. https://demo-cdn.photoprism.app/api/v1/t/63a6777ab09e6852817970a844667d8881a9a484/67cgcqrk/fit_2560 **(2048 x 1365)**
2. https://demo-cdn.photoprism.app/api/v1/t/63a6777ab09e6852817970a844667d8881a9a484/67cgcqrk/fit_1920 **(1800 x 1200)**
3. https://demo-cdn.photoprism.app/api/v1/t/63a6777ab09e6852817970a844667d8881a9a484/67cgcqrk/fit_1280 **(1280 x 853)**
4. https://demo-cdn.photoprism.app/api/v1/t/63a6777ab09e6852817970a844667d8881a9a484/67cgcqrk/fit_720 **(720 x 480)**

So it looks like the viewer loads a smaller version of the image as the window shrinks. That makes sense.

However when I resize the image at various sizes on my server, check out what I get: (Yes I know these links won't work for you)
1. http://192.168.0.126:2342/api/v1/t/beed83316f80a63d1fa02a36214e23810cdde51c/1yvlms1h/fit_2560 **(500 x 500)**
2. http://192.168.0.126:2342/api/v1/t/beed83316f80a63d1fa02a36214e23810cdde51c/1yvlms1h/fit_1920 **(500 x 500)**
3. http://192.168.0.126:2342/api/v1/t/beed83316f80a63d1fa02a36214e23810cdde51c/1yvlms1h/fit_1280 **(500 x 500)**
4. http://192.168.0.126:2342/api/v1/t/beed83316f80a63d1fa02a36214e23810cdde51c/1yvlms1h/fit_720 **(542 x 720)**

It's loading the thumbnail for the largest four images! The image shown in the new tab is the image cropped into a square!
And when I do the "open image in new tab" trick, the image is actually a square. When it shows the stretched/skewed image in the photo viewer, it takes the thumbnail and stretches it to he dimensions of the real photo. That's why it looks funny. Notice how the image in the photo viewer (left) shows the same stuff as the "Open image in new tab" view (right). It's just stretching the thumbnail to be the proper dimensions of the actual photo.

I'm not sure why the smallest version (fit_720) renders at the proper ratio.

![image](https://github.com/user-attachments/assets/18ce70dc-ed7f-46ad-9190-917754cdff3a)

I think this is progress! So why is the image scaler only pointing to the thumbnail image?

**Reply by srett** [reply]
That's odd indeed. Since your URLs go to port 2342 I assume you're accessing PP directly and not via some proxy that might mess things up.

Can you check what value the sliders under *Settings -> Advanced -> Preview Images / Image Quality* have? I'm not sure this could have any impact on this but it's the only idea I have right now. :)

**Reply by lastzero** [reply]
Definitely sounds like some problem with the cache/thumbnails folder or the thumbnail settings.

**Reply by ad08fee3** [reply]
Hi! Sorry for the late reply. I don't have email notifications enabled on github so I'm only remembering to check this thread every once in a while...

Here are my settings. I don't know if I have _ever_ tweaked these. So if they are not the defaults I am not sure how that came to be.
![image](https://github.com/user-attachments/assets/9d22aeda-ece4-4735-8975-9cebf50f38aa)

**Reply by ad08fee3** [reply]
Fascinating! 
I looked at [the advanced settings on the demo server](https://demo.photoprism.app/library/settings/advanced):
![image](https://github.com/user-attachments/assets/f81140e1-2362-4849-8631-2984cbf4b776)

To try to mimic these settings, on my own server I adjusted all the sliders down to the bottom. I'm not sure how the sliders were all set to zero on the demo server, as my server only let them as low as these values: 

![image](https://github.com/user-attachments/assets/c3ac673c-2e47-4ebb-88bb-60e47280d0bc)

But, after I changed this setting and restarted the server, the image previews started **working just fine!** At least, it did for images that I have not expanded before. If I go find a random image in the middle of the server and click on it, the preview renders the image properly instead of the skewed thumbnail. However, if I look at any picture I have previewed _before_ making those changes to the settings then I still see the skewed thumbnail in the preview instead. I clicked those example links I have posted in my previous comment and they still render as the skewed thumbnails as well.

So whatever settings I modified fixed the issue, however now I must somehowrefresh the cache of images that I have looked at before.

At least a quasi-workaround has been found. By adjusting the settings to what I have shown in this comment, the stretched-thumbnail issue goes away. However it doesn't fix the "previously-broken" previews.

I can confirm that when I re-disabled "Dynamic Previews" the problem came back.

So it seems like the sliders don't really matter at all, and it's purely an issue with Dynamic Previews. 

I am unable to disable Dynamic Previews on the Demo server, otherwise I would see if un-selecting that causes this bug to show up.

**Reply by lastzero** [reply]
Dynamic rendering will generate missing thumbs on demand. However, you can also (re-)generate them by manually running the "photoprism thumbs -f" command in a terminal. Alternatively, a full rescan will also  generate missing thumbs (but won't replace any existing thumbnail images eg if you changed the quality settings).

**Reply by lastzero** [reply]
Note that the Advanced Settings cannot be changed on our public demo for security reasons. That's why the related controls are disabled.

**Reply by srett** [reply]
> Here are my settings. I don't know if I have _ever_ tweaked these. So if they are not the defaults I am not sure how that came to be. ![image](https://private-user-images.githubusercontent.com/2447027/367153078-9d22aeda-ece4-4735-8975-9cebf50f38aa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYzMDg0ODEsIm5iZiI6MTcyNjMwODE4MSwicGF0aCI6Ii8yNDQ3MDI3LzM2NzE1MzA3OC05ZDIyYWVkYS1lY2U0LTQ3MzUtODk3NS05Y2ViZjUwZjM4YWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MTRUMTAwMzAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTcwNWQ0Y2JmYmY3YWFiNjg4OGE5OTZjZTgxZTBiNmM5NTcyN2IxNzFkNTFlYzViNDRiMzU4MjU2NDU2YzZjOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.lYFKzKCUi2lxIh319ClbFsk-dvgFFvU0ZYRLNGO_t4U)

FWIW, the only thing that looks off here is the static size limit, I have the others at 7680px too, and static size at 1920px. Either way it doesn't really explain the issue you were experiencing, but at least it seems to be fixed now.

I think the demo server simply doesn't expose the configured values, moving all the sliders to the lowest value is probably not what you want. It severely limits the maximum size of the downscaled images, and JPEG quality 25 should lead to terrible compression artifacts in thumbs and preview images. Maybe try increasing the values again and check if the issue comes back.

**Reply by ad08fee3** [reply]
For now I'm just running `photoprism thumbs -f` and leaving Dynamic Previews on :) 

Though this does make me wish that I could run a command that would just dump the thumbnails, rather than re-generating all of them. That defeats the purpose of dynamic previews, no?

**Reply by lastzero** [reply]
thumbs without -f will only generate missing static thumbs. See the command help which you can display with --help.

**Reply by ad08fee3** [reply]
Well in any case, the process completed and now all of my image previews are of the proper size and ratio. So I'd say this is fully mitigated! 

Still not sure of the root cause, but at least this thread now exists for future Google searches :)

**Reply by lastzero** [reply]
Typically this happens when there's a problem with the storage folder (eg wrongly or not mounted), you've restored an incomplete backup, or you have a broken configuration.

**Reply by ad08fee3** [reply]
I'm not sure how that could have happened though. I have had Photoprism up and running for well over a year, and this happened out of the blue. All I do is update the docker image regularly. That's about it.

**Comment by turastangras** [comment]
i am just about to start using photoprism as replacement for synology photos on my synology nas (ds918+).
i did a fresh install using docker (synology container manager) and portainer, using mostly the suggested presets from the provided templates. after indexing the first few folders, i have noticed as well, that some of my photos seem to be weirdly streched by the viewer.

root-cause could be something different, but that's what i found...

it seems that all photos that are displayed "stretched" in the viewer are "virtually rotated".
(i recognized that, because 2 examples had obviously a wrong virtual rotation of 90 degrees instead of 270 degrees and therefore are displayed upside-down now in the library views; which is correct)

so what i found:
- orientation field in eg win-explorer shows "rotate by 90 degree"
- all library views (and also eg Irfanview) are showing the photo correctly rotatet by 90 degress (in that case upside-down, which is correct).
- when using filters, the virtually rotated picture is not categorizes as being "portrait" (dimension of the picture is 2160x1440, which is indeed landscape. but since it needs to be rotated by 90 degree, it is in fact still a portrait picture)
- the viewer seems to rotate the picture first and then stretch it to the original (landscape) dimensions. which is leading to a weird streched view.

samples attached (gui in german...)
br werner

---

[samples.zip](https://github.com/user-attachments/files/17443870/samples.zip)

**Reply by lastzero** [reply]
Is this also the case on [demo.photoprism.app](https://demo.photoprism.app/)?

Checking the file metadata with Exiftool may give a hint:

```
exiftool -n -j file.jpg
```

You are welcome to post the output here as follows so that we and others can give you feedback:

\`\`\`json
[{
  "SourceFile": "file.jpg",
  "ExifToolVersion": 12.76,
  ...
}]
\`\`\`

**Reply by lastzero** [reply]
Our troubleshooting guide now includes instructions on how to use Exiftool:
- https://docs.photoprism.app/getting-started/troubleshooting/metadata/

**Reply by turastangras** [reply]
hello and sorry for the late reply.

i found that for affected pics, xmp-metadata was set, but exif-data was missing.
although rotation info was present in xmp-metadata this obviously led to different interpretations with different software.
correcting that also solved my problem with these pics.

(in the meantime i am a silver subscriber and also learned docker compose syntax to efficiently setup multiple instances on my synology nas. so far i do like photoprism ;) btw... i needed to update synology-firewall with the ip-range of the default docker network. since that is not mentioned anywhere and i wasn't bright enough to realize that at first, this was the biggest problem in setting up these instances... although running, mariadb didn't accept connections from photoprism... maybe worth a doc-sidenote for new synology-photoprism-admins :D)

thx werner
